src/Microsoft.AspNet.OData/ODataOptions.cs

```diff
@@ -0,0 +1,23 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataOptions.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNet.OData.Query;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Provides programmatic configuration for the OData service.
+    /// </summary>
+    public class ODataOptions
+    {
+        /// <summary>
+        /// Gets or sets a value indicating if <see cref="ODataQueryOptions"/> should be reused between
+        /// <see cref="ODataQueryParameterBindingAttribute"/> and <see cref="EnableQueryAttribute"/>.
+        /// </summary>
+        public bool ParseODataQueryOptionsOnce { get; set; }
```

Comment:
>ParseODataQueryOptionsOnce [](http://example.com/codeflow?start=20&length=26)

In what scenarios would I *not* want this to be true? 

Is this there to protect backward compatibility, so that *if* someone happens to have code that modifies the query options (i.e., during Apply) without copying them first, that it doesn't affect other uses of QueryOptions?

In a future breaking change release, would we want to remove this option and always re-use the ODataQueryOptions?

Reply:
Personally I think it would ideally always be true, but for safety I made it opt in. I'd be fine with a break change release inverting this logic (i.e. default to `true` and but allow override to `false`).

Reply:
Since I've switched to `CompatibilityOptions` the change will now be active by default.